gameoflife.py

```diff
@@ -76,25 +95,25 @@ def print_board(board):
         print('|')
 
 
-def main(args):
-    if len(args) != 2:
-        print("Please enter 2 arguments\n\t-file path to initial board state\n\t-number of generations")
-        exit()
+def main(args : argparse.Namespace) ->None:
     try:
-        board = init_board(args[0])
-        generations = int(args[1])
-    except ValueError:
-        print(f"2nd argument must be a number, '{args[1]}' was entered.")
-        exit()
+        board = init_board(args.file_path)
     except Exception as e:
         print(e)
         exit()
 
+    generations = int(args.generations)
```

Comment:
no need to cast to `int`, because you told argparse to treat the `generation` parameter as `int`